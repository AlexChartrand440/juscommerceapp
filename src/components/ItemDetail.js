import React from 'react';
import styled from 'styled-components';
import { Image, View } from 'react-native';

import Button from './Button';

const Title = styled.Text`
  font-size: 18;
  color: #202020;
  letter-spacing: 0.5;
`;

const SubTitle = styled.Text`
  font-size: 14;
  color: #444444;
  font-weight: 200;
  letter-spacing: 1;
`;

const DetailsView = styled.View`
  flex-direction: row;
`;

const PriceText = styled.Text`
  color: #ff685f;
  font-size: 26;
  font-weight: 500;
  padding-bottom: 12;
`;

const AmountText = styled.Text`
  font-size: 12;
`;
const DetailField = styled.Text`
  font-size: 12;
  color: #444444;
  font-weight: 200;
`;
const DetailText = styled.Text`
  font-size: 16;
  color: #444444;
  line-height: 20;
`;
const ItemImageView = styled.View`
  padding-vertical: 16;
`;
const ItemImage = styled(Image)`
  width: 250;
  height: 200;
  border-radius: 5;
  align-self: center;
`;

const ButtonView = styled.View`
  align-self: center;
  padding-top: 16;
`;

export default ({ item, onSelect }) => (
  <View>
    <Title>{item.name}</Title>
    <SubTitle>{item.description}</SubTitle>
    <ItemImageView>
      <ItemImage source={{ uri: item.image }} />
    </ItemImageView>
    <DetailsView>
      <PriceText>{`R$${item.price}`}</PriceText>
      <AmountText>{item.amount}</AmountText>
    </DetailsView>
    <DetailField>Descrição</DetailField>
    <DetailText>{item.description}</DetailText>
    <ButtonView>
      <Button label='Adicionar ao carrinho' onPress={onSelect} />
    </ButtonView>
  </View>
);
