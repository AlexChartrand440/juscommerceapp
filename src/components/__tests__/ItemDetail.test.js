import React from 'react';
import renderer from 'react-test-renderer';
import ItemDetail from '../ItemDetail';

const product = {
  id: 1,
  name: 'test product1',
  description: 'test description 1',
  image:
    'https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681',
  price: 123
};

test('test product detail', () => {
  const wrapper = renderer.create(<ItemDetail item={product} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
