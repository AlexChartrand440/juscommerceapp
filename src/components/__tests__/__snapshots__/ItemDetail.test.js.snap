// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test product detail 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#202020",
          "fontSize": 18,
          "letterSpacing": 0.5,
        },
      ]
    }
  >
    test product1
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#444444",
          "fontSize": 14,
          "fontWeight": "200",
          "letterSpacing": 1,
        },
      ]
    }
  >
    test description 1
  </Text>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 16,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderRadius": 5,
            "height": 200,
            "width": 250,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ff685f",
            "fontSize": 26,
            "fontWeight": "500",
            "paddingBottom": 12,
          },
        ]
      }
    >
      R$123
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#444444",
          "fontSize": 12,
          "fontWeight": "200",
        },
      ]
    }
  >
    Descrição
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#444444",
          "fontSize": 16,
          "lineHeight": 20,
        },
      ]
    }
  >
    test description 1
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "paddingTop": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#ff685f",
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "height": 45,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0,
          "width": 250,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                  Object {
                    "fontFamily": "HelveticaNeue-Medium",
                  },
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "width": "100%",
                  },
                ]
              }
            >
              Adicionar ao carrinho
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
